### PR TITLE
[Feature] add training recipe and trained weights of Swin Transformer

### PR DESCRIPTION
**Pull request by wcrzlh**
Thank you for your contribution to the MindCV repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

The training recipe and trained weights of swin_tiny model are added. 

## Test Plan

Please refer to 'config/swintransformer/README.md' for reproducing and testing.

## Related Issues and PRs

Related issue: https://github.com/mindspore-lab/mindcv/issues/413

### CODE REVIEW

File: configs/swintransformer/swin_tiny_ascend.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# system
+mode: 0
+distribute: True
+num_parallel_workers: 8
+val_while_train: True
+val_interval: 1
+
+# dataset
+dataset: "imagenet"
+data_dir: "path/to/imagenet"
```
Comment (Songyuanwei) [inline configs/swintransformer/swin_tiny_ascend.yaml:10]: "path/to/imagenet" -> "/path/to/imagenet"

File: configs/swintransformer/swin_tiny_ascend.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+dataset_sink_mode: True
+amp_level: "O2"
+
+# loss
+loss: "CE"
+loss_scale: 1024.0
+label_smoothing: 0.1
+
+# lr scheduler
+scheduler: "warmup_cosine_decay"
```
Comment (Songyuanwei) [inline configs/swintransformer/swin_tiny_ascend.yaml:48]: 参数更正 "warmup_cosine_decay" -> "cosine_decay"

File: configs/swintransformer/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+### Training
+<!--- Guideline: Please avoid using shell scripts in the command line. Python scripts preferred. -->
+
+* Distributed Training
+
+It is easy to reproduce the reported results with the pre-defined training recipe. For distributed training on multiple Ascend 910 devices, please run
+
+```shell
+# distrubted training on multiple GPU/Ascend devices
```
Comment (Songyuanwei) [inline configs/swintransformer/README.md:66]: distrubted -> distributed

File: configs/swintransformer/swin_tiny_ascend.yaml (hunk 1 above)
Comment (wcrzlh) [reply configs/swintransformer/swin_tiny_ascend.yaml]: fixed

File: configs/swintransformer/swin_tiny_ascend.yaml (hunk 2 above)
Comment (wcrzlh) [reply configs/swintransformer/swin_tiny_ascend.yaml]: fixed

File: configs/swintransformer/README.md (hunk 3 above)
Comment (wcrzlh) [reply configs/swintransformer/README.md]: fixed

**Review by geniuspatrick — APPROVED** [review]

**Comment by geniuspatrick** [comment]
closed #413

**Pull request merged**